ReflectometryServer/server_status_manager.py

```diff
@@ -193,6 +194,15 @@ def update_error_log(self, message: str, exception: Optional[Exception] = None):
             logger.exception(message, exc_info=exception)
         else:
             logger.error(message)
+
+        # If adding the new message exceeds the maximum character size of the error log,
+        # the oldest message is removed until the log size is under the maximum.
+        # One is added or removed from the size to compensate for the new line (check "_error_log_as_string").
+        MAX_ERROR_LOG_SIZE = 10_000
+        self._error_log_character_size += (len(message) + 1)
+        while self._error_log_character_size > MAX_ERROR_LOG_SIZE:
+            self._error_log_character_size -= (len(self._error_log.pop(0)) + 1)
+
```

Comment:
I believe pop(0) on a normal python list can be inefficient, is it worth investigating `dequeue`  ?

Reply:
potentially you could create you own error log class with the `dequeue` which is constructed with a character limit parameter and then handles all the other stuff itself?